### DISCUSSION TITLE
Allow props that intialize store to udpate

### DISCUSSION THREAD

**Discussion by bragaru-i** [pmndrs/zustand#1417] [Q&A] [answered]
First, thanks for the great package!
Do we have the possibility to update initialized props for store?
Currently, if the external prop is updated, store stays the same.
Codesandbox:
https://codesandbox.io/s/focused-archimedes-cyrxq7
```ts
export default function App() {
  const [bears, setBears] = useState(0);

  useEffect(() => {
    const timeoutId = setTimeout(() => setBears(10), 3000);
    return () => clearTimeout(timeoutId);
  }, []);
  console.log(bears);
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>
      <BearProvider bears={bears}>
        <CommonConsumer />
      </BearProvider>
    </div>
  );
}
```

`bears` from the `BearStore` will be 0, not 10 after a timeout.
Is there a possibility to update the initialized value?

**Comment by dai-shi** [comment, marked as answer]
I'm not sure if it's best practice to update store from props, but you can do that with `useEffect` if you want to.

**Comment by bragaru-i** [comment]
@dai-shi, the issue is that we have a composed state in a player [app](https://github.com/Collaborne/video-player/blob/669af656ac563efd65dbbe11b812a90a6c05c38a/src/store/media-store.ts#L333-L343).
and here, we use  a slicer  for external [props](https://github.com/Collaborne/video-player/blob/669af656ac563efd65dbbe11b812a90a6c05c38a/src/store/media-store.ts#L24-L27) 
the "props slice" can change when we get the file type, to display corresponding UI(audio vs video).

**Reply by dai-shi** [reply]
Does it only apply with slice pattern? If you need further discussion, please create a minimal reproduction with codesandbox https://react.new.
